src/simplewallet/simplewallet.cpp

```diff
@@ -1395,50 +1515,42 @@ bool simple_wallet::import_multisig_main(const std::vector<std::string> &args, b
   std::vector<cryptonote::blobdata> info;
   for (size_t n = 0; n < args.size(); ++n)
   {
-    if (called_by_mms)
-    {
-      info.push_back(args[n]);
-    }
-    else
+    const std::string &filename = args[n];
+    std::string data;
+    bool r = m_wallet_impl->loadFromFile(filename, data);
+    if (!r)
     {
-      const std::string &filename = args[n];
-      std::string data;
-      bool r = m_wallet->load_from_file(filename, data);
-      if (!r)
-      {
-        fail_msg_writer() << tr("failed to read file ") << filename;
-        return false;
-      }
-      info.push_back(std::move(data));
+      fail_msg_writer() << tr("failed to read file ") << filename;
+      return false;
     }
+    info.push_back(std::move(data));
   }
 
   SCOPED_WALLET_UNLOCK_ON_BAD_PASSWORD(return false;);
 
   // all read and parsed, actually import
-  try
   {
     m_in_manual_refresh.store(true, std::memory_order_relaxed);
     epee::misc_utils::auto_scope_leave_caller scope_exit_handler = epee::misc_utils::create_scope_leave_handler([&](){m_in_manual_refresh.store(false, std::memory_order_relaxed);});
-    size_t n_outputs = m_wallet->import_multisig(info);
+    size_t n_outputs = m_wallet_impl->importMultisigImages(info);
+
+    m_wallet_impl->statusWithErrorString(error_code, error_msg);
+    if (error_code)
+    {
+      fail_msg_writer() << tr("Failed to import multisig info: ") << error_msg;
+      return false;
+    }
+
     // Clear line "Height xxx of xxx"
```

Comment:
What the heck is this, and more importantly, is it still revelant after switching to `importMultisigImages`?